2.1.2 missing link against pthreads

Hi! When building 2.1.2 for Arch Linux, I ran into issues in the linking stage. It seems as if yoshimi is not linking against pthreads:

```
[ 82%] Linking CXX executable yoshimi
/usr/bin/cmake -E cmake_link_script CMakeFiles/yoshimi.dir/link.txt --verbose=1
/usr/bin/c++ -ffast-math -fomit-frame-pointer -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O3 -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -rdynamic CMakeFiles/yoshimi.dir/Interface/InterChange.cpp.o CMakeFiles/yoshimi.dir/Interface/Data2Text.cpp.o CMakeFiles/yoshimi.dir/Interface/Text2Dat
a.cpp.o CMakeFiles/yoshimi.dir/Interface/MidiLearn.cpp.o CMakeFiles/yoshimi.dir/Interface/MidiDecode.cpp.o CMakeFiles/yoshimi.dir/CLI/CmdInterface.cpp.o CMakeFiles/yoshimi.dir/CLI/CmdInterpreter.cpp.o CMakeFiles/yoshimi.dir/Misc/CmdOptions.cpp.o CMakeFiles/yoshimi.dir/Misc/Config.cpp.o CMakeFiles/yoshimi.dir/Misc/Synt
hEngine.cpp.o CMakeFiles/yoshimi.dir/Misc/Bank.cpp.o CMakeFiles/yoshimi.dir/Misc/Splash.cpp.o CMakeFiles/yoshimi.dir/Misc/Microtonal.cpp.o CMakeFiles/yoshimi.dir/Misc/Part.cpp.o CMakeFiles/yoshimi.dir/Misc/XMLwrapper.cpp.o CMakeFiles/yoshimi.dir/Misc/WavFile.cpp.o CMakeFiles/yoshimi.dir/Params/ADnoteParameters.cpp.o C
MakeFiles/yoshimi.dir/Params/EnvelopeParams.cpp.o CMakeFiles/yoshimi.dir/Params/FilterParams.cpp.o CMakeFiles/yoshimi.dir/Params/LFOParams.cpp.o CMakeFiles/yoshimi.dir/Params/SUBnoteParameters.cpp.o CMakeFiles/yoshimi.dir/Params/PADnoteParameters.cpp.o CMakeFiles/yoshimi.dir/Params/Controller.cpp.o CMakeFiles/yoshimi.
dir/Params/Presets.cpp.o CMakeFiles/yoshimi.dir/Params/PresetsStore.cpp.o CMakeFiles/yoshimi.dir/Params/UnifiedPresets.cpp.o CMakeFiles/yoshimi.dir/Params/OscilParameters.cpp.o CMakeFiles/yoshimi.dir/Synth/ADnote.cpp.o CMakeFiles/yoshimi.dir/Synth/Envelope.cpp.o CMakeFiles/yoshimi.dir/Synth/LFO.cpp.o CMakeFiles/yoshim
i.dir/Synth/OscilGen.cpp.o CMakeFiles/yoshimi.dir/Synth/SUBnote.cpp.o CMakeFiles/yoshimi.dir/Synth/Resonance.cpp.o CMakeFiles/yoshimi.dir/Synth/PADnote.cpp.o CMakeFiles/yoshimi.dir/DSP/FFTwrapper.cpp.o CMakeFiles/yoshimi.dir/DSP/AnalogFilter.cpp.o CMakeFiles/yoshimi.dir/DSP/FormantFilter.cpp.o CMakeFiles/yoshimi.dir/D
SP/SVFilter.cpp.o CMakeFiles/yoshimi.dir/DSP/Filter.cpp.o CMakeFiles/yoshimi.dir/DSP/Unison.cpp.o CMakeFiles/yoshimi.dir/Effects/Alienwah.cpp.o CMakeFiles/yoshimi.dir/Effects/Chorus.cpp.o CMakeFiles/yoshimi.dir/Effects/Echo.cpp.o CMakeFiles/yoshimi.dir/Effects/EffectLFO.cpp.o CMakeFiles/yoshimi.dir/Effects/EffectMgr.c
pp.o CMakeFiles/yoshimi.dir/Effects/Effect.cpp.o CMakeFiles/yoshimi.dir/Effects/Phaser.cpp.o CMakeFiles/yoshimi.dir/Effects/Reverb.cpp.o CMakeFiles/yoshimi.dir/Effects/EQ.cpp.o CMakeFiles/yoshimi.dir/Effects/Distorsion.cpp.o CMakeFiles/yoshimi.dir/Effects/DynamicFilter.cpp.o CMakeFiles/yoshimi.dir/MusicIO/MusicClient.
cpp.o CMakeFiles/yoshimi.dir/MusicIO/MusicIO.cpp.o CMakeFiles/yoshimi.dir/MusicIO/JackEngine.cpp.o CMakeFiles/yoshimi.dir/MusicIO/AlsaEngine.cpp.o CMakeFiles/yoshimi.dir/PresetsUI.cpp.o CMakeFiles/yoshimi.dir/EnvelopeUI.cpp.o CMakeFiles/yoshimi.dir/LFOUI.cpp.o CMakeFiles/yoshimi.dir/FilterUI.cpp.o CMakeFiles/yoshimi.d
ir/VirKeyboardUI.cpp.o CMakeFiles/yoshimi.dir/ConfigUI.cpp.o CMakeFiles/yoshimi.dir/SUBnoteUI.cpp.o CMakeFiles/yoshimi.dir/ResonanceUI.cpp.o CMakeFiles/yoshimi.dir/OscilGenUI.cpp.o CMakeFiles/yoshimi.dir/ADnoteUI.cpp.o CMakeFiles/yoshimi.dir/PADnoteUI.cpp.o CMakeFiles/yoshimi.dir/EffUI.cpp.o CMakeFiles/yoshimi.dir/Ban
kUI.cpp.o CMakeFiles/yoshimi.dir/PartUI.cpp.o CMakeFiles/yoshimi.dir/MicrotonalUI.cpp.o CMakeFiles/yoshimi.dir/MasterUI.cpp.o CMakeFiles/yoshimi.dir/MasterMiscUI.cpp.o CMakeFiles/yoshimi.dir/ParametersUI.cpp.o CMakeFiles/yoshimi.dir/ConsoleUI.cpp.o CMakeFiles/yoshimi.dir/VectorUI.cpp.o CMakeFiles/yoshimi.dir/MidiLearn
UI.cpp.o CMakeFiles/yoshimi.dir/UI/DynamicTooltip.cpp.o CMakeFiles/yoshimi.dir/UI/WidgetPDial.cpp.o CMakeFiles/yoshimi.dir/UI/WidgetCheckButton.cpp.o CMakeFiles/yoshimi.dir/UI/WidgetSpinner.cpp.o CMakeFiles/yoshimi.dir/UI/WidgetMWSlider.cpp.o CMakeFiles/yoshimi.dir/UI/YoshiWin.cpp.o CMakeFiles/yoshimi.dir/UI/MiscGui.c
pp.o CMakeFiles/yoshimi.dir/main.cpp.o -o yoshimi  -lfontconfig -lfreetype -lfltk_images -lfltk_forms -lfltk_gl -lGL -lfltk -lX11 -lXext -lm -lmxml -lasound -ljack -lfftw3f -lcairo -lncurses -lform -lreadline -lz -ldl
/usr/bin/ld: CMakeFiles/yoshimi.dir/Interface/InterChange.cpp.o: undefined reference to symbol 'sem_post@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/yoshimi.dir/build.make:1522: yoshimi] Error 1
make[2]: Leaving directory '/build/yoshimi/src/yoshimi-2.1.2/build'
make[1]: *** [CMakeFiles/Makefile2:103: CMakeFiles/yoshimi.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

I think you probably want to use this (and bump your required cmake version):
https://cmake.org/cmake/help/latest/module/FindThreads.html?highlight=threads#module:FindThreads

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -139,6 +139,7 @@ endif ()
 
 find_package (PkgConfig REQUIRED)
 
+find_package(Threads REQUIRED)
 
 check_c_source_compiles (
     "#include <argp.h>
@@ -435,6 +436,7 @@ set(ExternLibraries
     ${Readline_LIBRARY}
     ${ZLIB_LIBRARIES}
     ${LIBDL_LINUX}
+    ${CMAKE_THREAD_LIBS_INIT}
 )
 
 if ("${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")

